src/contracts/services/derivativeScrt.ts

```diff
@@ -0,0 +1,273 @@
+ */
+async function queryDerivativeScrtStakingInfo({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) {
+  return lastValueFrom(queryDerivativeScrtStakingInfo$({
+    contractAddress,
+    codeHash,
+    lcdEndpoint,
+    chainId,
+  }));
+}
+
+/**
+ * parse the response for fee info
+ */
+const parseDerivativeScrtFeeInfo = (
+  response: DerivativeScrtFeeInfoResponse,
+): DerivativeScrtFeeInfo => ({
+  withdrawFee: Number(response.fee_info.withdraw) / 100000,
+  depositFee: Number(response.fee_info.deposit) / 100000,
+});
+
+/**
+ * parse the response from the batch query contract
+ */
+const parseDerivativeScrtAllInfo = (
+  response: BatchQueryParsedResponse,
+): DerivativeScrtInfo => {
+  const stakingInfoResponse = response.find(
+    (nextBatchItem: BatchQueryParsedResponseItem) => nextBatchItem.id === 'staking_info',
+  );
+  const feeInfoResponse = response.find(
+    (nextBatchItem: BatchQueryParsedResponseItem) => nextBatchItem.id === 'fee_info',
+  );
+  if (!stakingInfoResponse || !feeInfoResponse) {
+    throw new Error(`Unable to parse batch query response: ${response}`);
+  }
+  return {
+    ...parseDerivativeScrtStakingInfo(stakingInfoResponse.response),
+    ...parseDerivativeScrtFeeInfo(feeInfoResponse.response),
+  };
+};
+
+/**
+ * query the fee info
+ */
+const queryDerivativeScrtFeeInfo$ = ({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) => getActiveQueryClient$(lcdEndpoint, chainId).pipe(
+  switchMap(({ client }: { client: any }) => sendSecretClientContractQuery$({
+    queryMsg: msgQueryScrtDerivativeFees(),
+    client,
+    contractAddress,
+    codeHash,
+  })),
+  map((response: any) => parseDerivativeScrtFeeInfo(response as DerivativeScrtFeeInfoResponse)),
+  first(),
+);
+
+/**
+ * query the fee info
+ */
+async function queryDerivativeScrtFeeInfo({
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  contractAddress: string,
+  codeHash?: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) {
+  return lastValueFrom(queryDerivativeScrtFeeInfo$({
+    contractAddress,
+    codeHash,
+    lcdEndpoint,
+    chainId,
+  }));
+}
+
+/**
+ * query both the staking info and the fee info
+ */
+const queryDerivativeScrtAllInfo$ = ({
+  queryRouterContractAddress,
+  queryRouterCodeHash,
+  contractAddress,
+  codeHash,
+  lcdEndpoint,
+  chainId,
+}: {
+  queryRouterContractAddress: string,
+  queryRouterCodeHash?: string,
+  contractAddress: string,
+  codeHash: string,
+  lcdEndpoint?: string,
+  chainId?: string,
+}) => batchQuery$({
+  queries: [
+    {
+      id: 'staking_info',
+      contract: {
+        address: contractAddress,
+        codeHash,
+      },
+      queryMsg: msgQueryScrtDerivativeStakingInfo(Math.round(new Date().getTime() / 1000)),
```

Comment:
to add flexibility to this service, we should add an optional time parameter as a function input. I don't fully understand even what time does and why it's a message input. Does it allow you to get historical data points? We should add this into the docs to explain to users what the input is and why it's optional.

Reply:
Dude I don't even understand, I don't think it's optional tho

Reply:
i didn't mean that it is optional, i meant that it would be optional as a parameter if we were to make it a parameter because we would have a default of Date().getTime()

If there truly is no value to ever adjusting the time, then we can skip the param. My assumption was that it does something, otherwise why wouldn't we just have made the query without this input?